File: online judge.kaner murillo/c++/black list/679.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int depth, label, level, act, pre, i, n, cases;
    cin >> cases;

    while (cases != -1) {
        vector<bool> flags(524288, false);
        vector<bool> flags2(524288, -1);

        for (i = 0; i < cases; i++) {
            
            cin >> depth >> label;

            for (n = 0; n < label; n++) {
                level = 1;
                act = 1;
                pre = -1;

                while (level != depth) {
                    pre = act;

                    if (!flags[act] || flags2[act] != i) {
                        act = act * 2;
                    } else {
                        act = act * 2 + 1;
                    }

                    flags[pre] = !flags[pre];
                    flags2[pre] = i;
                    level++;
                }
            }

            cout << act << endl;
        }

        cin >> cases;
    }

    return 0;
}

File: online judge.kaner murillo/python/679.py
#https://onlinejudge.org/external/6/679.pdf
#679

from sys import stdin


def main():

    cases = int(stdin.readline())

    while cases != -1:


        for i in range(cases):

            depth, label = [int(x) for x in stdin.readline().split()]
            flags = [False for x in range(524288)]

            for n in range(label):

                level = 1
                act = 1
                pre = None

                while level != depth:

                    pre = act

                    if not flags[act]:

                        act = act*2

                    else:

                        act = act*2+1 

                    flags[pre] = not flags[pre]
                    level+=1

            print(act)

        cases = int(stdin.readline())
main()